ServerCore/Pages/Puzzles/SubmitFeedback.cshtml.cs

```diff
@@ -44,15 +55,33 @@ public async Task<IActionResult> OnPostAsync(int puzzleId)
                 return Page();
             }
 
-            Feedback.Puzzle = await _context.Puzzles.Where(m => m.ID == puzzleId).FirstOrDefaultAsync();
-
-            if (Feedback.Puzzle == null) 
-            { 
-                return NotFound(); 
-            } 
+            Feedback editableFeedback = await _context.Feedback.Where((f) => f.Puzzle.ID == puzzleId && f.Submitter == LoggedInUser).FirstOrDefaultAsync();
+            if (editableFeedback == null)
+            {
+                Feedback.SubmissionTime = DateTime.UtcNow;
+                Feedback.Submitter = LoggedInUser;
+                Feedback.Puzzle = await _context.Puzzles.Where(m => m.ID == puzzleId).FirstOrDefaultAsync();
+                if (Feedback.Puzzle == null)
+                {
+                    return NotFound();
```

Comment:
Nit: Is it possible to add text to the NotFound constructor to indicate what was not found?

Reply:
Added